### ISSUE TITLE
pylint finding W101 error on commented out code

### ISSUE THREAD

**Issue by panewt** [pylint-dev/pylint#10198]
### Bug description

```python
#!/usr/bin/python3 -B

def main():
    theMsg = 'a message we want to use for a RuntimeError'
    raise RuntimeError (theMsg)
    """
    some code that pylint shouldn't be scanning
    """

if __name__ == '__main__':
    main()
```

### Configuration

```ini

```

### Command used

```shell
pylint-3 per rocky9
```

### Pylint output

```python
[paul@grotto python]$ pylint CENSUS_household.py 
running pylint-3
************* Module CENSUS_household
CENSUS_household.py:2432:3: W0101: Unreachable code (unreachable)
CENSUS_household.py:2441:3: W0101: Unreachable code (unreachable)

------------------------------------------------------------------
Your code has been rated at 9.99/10 (previous run: 9.99/10, +0.00)

[paul@grotto python]$
```

### Expected behavior

The line pylint is barking on is a comment and therefore it should care whether code flow treats this line as unreachable

Note that I am not running the most current versions of Python and pylint as I only have what RHEL/rocky 9.5 will give me. So if you have already corrected this, I apologize for the noise

### Pylint version

```shell
[paul@grotto python]$ pylint --version
running pylint-3
pylint 2.13.7
astroid 2.11.6
Python 3.9.21 (main, Dec  5 2024, 00:00:00) 
[GCC 11.5.0 20240719 (Red Hat 11.5.0-2)]
[paul@grotto python]$
```

### OS / Environment

Rocky Linux release 9.5 (Blue Onyx)

### Additional dependencies

```python

```

**Comment by Pierre-Sassoulas** [comment]
I think pylint is working as intended this is not a comment but a string declaration that is not assigned to a variable ? (A comment should be `# something`). Docstrings should not be in the middle of the code but at the module beginning and after function declaration.

**Comment by Pierre-Sassoulas** [comment]
Closing following discussion in #10187
